contrib/endpoints/src/api_manager/context/request_context.cc

```diff
@@ -225,6 +225,39 @@ void RequestContext::FillCheckRequestInfo(
   info->allow_unregistered_calls = method()->allow_unregistered_calls();
 }
 
+// TODO:: JAEBONG:: All hard corded for test. Need to read it from the
+// configuration
+void RequestContext::FillAllocateQuotaRequestInfo(
+    service_control::QuotaRequestInfo *info) {
+  FillOperationInfo(info);
+
+  info->method_name = this->method_call_.method_info->name();
+
+  // TODO:: JAEBONG::
+  info->labels["servicecontrol.googleapis.com/caller_ip"] =
+      request_->GetClientIP();
+  info->labels["servicecontrol.googleapis.com/referer"] = this->http_referer_;
+  info->labels["servicecontrol.googleapis.com/user"] = "integration_test_user";
```

Comment:
Usually don't use map in Info struct.   Info should be just a flat struct  with all fields listed.

in the service_control/proto.c   FillQuotaRequest( info,  *pb)  that will create labels based on the fields

It will be good if we can do that same for metrics.   

But I understand quota_metics may need a vector or map since it is from service config.